pkg/util/archive/zip_test.go

```diff
@@ -0,0 +1,179 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package archive
+
+import (
+	"archive/zip"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestZip_WrongDestinationFile(t *testing.T) {
+	err := Zip([]string{"a"}, "destination.txt")
+
+	assert.Error(t, err, "must have a .zip extension")
+}
+
+func TestZip_ExtistingDestuination(t *testing.T) {
+	tmpDir := t.TempDir()
+	tempLocation := filepath.Join(tmpDir, "destination.zip")
+	_, err := os.Create(tempLocation)
+	assert.Nil(t, err)
+
+	err = Zip([]string{"a"}, tempLocation)
+
+	assert.Error(t, err, "file already exists:")
+}
+
+func TestZip_DoNotZipSymlinks(t *testing.T) {
+	tmpDestinationDir := t.TempDir()
+	zipTempLocation := filepath.Join(tmpDestinationDir, "destination.zip")
```

Comment:
Not a blocker I feel like a lot of these tests have some common code that could be deduped